sample-common/src/main/java/com/beepiz/blegattcoroutines/sample/common/MainViewModel.kt

```diff
@@ -22,7 +23,7 @@ class MainViewModel : ScopedViewModel() {
     fun logNameAndAppearance(deviceMacAddress: String = defaultDeviceMacAddress,
                              connectionTimeoutInMillis: Long = 5000L) {
         operationAttempt?.cancel()
-        operationAttempt = launch {
+        operationAttempt = GlobalScope.launch {
```

Comment:
At first, I was wondering why you switched to `GlobalScope` while the class you're in is already a `CoroutineScope` that gets cancelled properly when `onCleared()` is called… then I understood why:
The backing `Job` from `ScopedViewModel` is from a regular `Job()` call, which means that once you cancel a child, you can never launch successfully any additional coroutine on it, it will get cancelled ASAP.

The proper fix is to use `SupervisorJob()` in `ScopedViewModel`. Again, this is unrelated to Kotlin 1.3 migration, but I thank you for making me aware of this issue that slipped by. I'll work on this.

Reply:
Welcome 👍